google-cloud-bigquery/lib/google/cloud/bigquery/policy.rb

```diff
@@ -0,0 +1,341 @@
+        #     this value reverts to `serviceAccount:<emailid>` and the undeleted service account retains the role in
+        #     the binding.
+        #   * `deleted:group:<emailid>?uid=<uniqueid>`: An email address (plus unique identifier) representing a Google
+        #     group that has been recently deleted. For example, `admins@example.com?uid=123456789012345678901`. If the
+        #     group is recovered, this value reverts to `group:<emailid>` and the recovered group retains the role in
+        #     the binding.
+        #   * `domain:<domain>`: The G Suite domain (primary) that represents all the users of that domain. For example,
+        #     `google.com` or `example.com`.
+        #
+        # @return [Binding, nil] The binding object that was added to the policy. This object is mutable and may be used
+        #   to add or remove members within the context of a block passed to {Table.update_policy}.
+        #
+        # @example Update a mutable policy with {Table#update_policy}.
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table"
+        #
+        #   table.update_policy do |p|
+        #     p.grant members: "user:viewer@example.com", role: "roles/viewer"
+        #   end # 2 API calls
+        #
+        def grant members:, role:
+          new_binding = Binding.new role, members
+          if (i = @bindings.find_index { |b| b.role == role })
+            @bindings[i] = new_binding
+          else
+            @bindings << new_binding
+          end
+          new_binding
+        end
+
+        ##
+        # Convenience method deleting a binding in the policy. Returns `nil` if no binding is present for the role. See
+        # [Understanding Roles](https://cloud.google.com/iam/docs/understanding-roles) for a list of primitive and
+        # curated roles. See [BigQuery Table ACL
+        # permissions](https://cloud.google.com/bigquery/docs/table-access-controls-intro#permissions) for a list of
+        # values and patterns for members.
+        #
+        # @param [String] role A role that is bound to members in the policy. For example, `roles/viewer`,
+        #   `roles/editor`, or `roles/owner`. Required.
+        #
+        # @return [Binding, nil] The frozen binding object, or `nil` if no binding exists for the given role.
+        #
+        # @example Update a mutable policy with {Table#update_policy}.
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table"
+        #
+        #   table.update_policy do |p|
+        #     p.revoke role: "roles/owner"
+        #   end # 2 API calls
+        #
+        def revoke role:
```

Comment:
LGTM so far. We need to be able to support `members:` too.

Reply:
Done.